shared/src/main/kotlin/com/egm/stellio/shared/queryparameter/AllowedParameters.kt

```diff
@@ -0,0 +1,69 @@
+package com.egm.stellio.shared.queryparameter
+
+import jakarta.validation.Constraint
+import jakarta.validation.ConstraintValidator
+import jakarta.validation.ConstraintValidatorContext
+import org.springframework.http.HttpStatus
+import org.springframework.util.MultiValueMap
+import kotlin.reflect.KClass
+
+@Target(
+    AnnotationTarget.VALUE_PARAMETER,
+)
+@Retention(
+    AnnotationRetention.RUNTIME
+)
+@Constraint(validatedBy = [ AllowedParameters.ParamValidator::class])
+annotation class AllowedParameters(
+    val implemented: Array<QueryParameter> = [],
+    val notImplemented: Array<QueryParameter> = [],
+    val message: String = "Invalid parameter received",
+    val groups: Array<KClass<*>> = [],
+    val payload: Array<KClass<*>> = [],
+) {
+    class ParamValidator :
+        ConstraintValidator<AllowedParameters, MultiValueMap<String, String>> {
+        private var implemented: List<String> = listOf()
+        private var notImplemented: List<String> = listOf()
+
+        override fun initialize(allowedParameters: AllowedParameters) {
+            this.implemented = allowedParameters.implemented.map(QueryParameter::key)
+            this.notImplemented = allowedParameters.notImplemented.map(QueryParameter::key)
+        }
+
+        override fun isValid(params: MultiValueMap<String, String>, context: ConstraintValidatorContext): Boolean {
+            if (implemented.containsAll(params.keys)) {
+                return true
+            }
+
+            val notImplementedKeys = params.keys.intersect(notImplemented)
+            val errorKeys = params.keys - notImplementedKeys - implemented
+
+            context.disableDefaultConstraintViolation()
+
+            val message = StringBuilder().apply {
+                if (notImplementedKeys.isNotEmpty()) {
+                    append(
+                        "The '${notImplementedKeys.joinToString("', '")}' parameter is not implemented yet. "
```

Comment:
in plural maybe? can we write it like this in English: `The ... parameter(s) is(are) not implemented yet`?